src/main/java/io/github/a5h73y/parkour/listener/PlayerMoveListener.java

```diff
@@ -37,11 +38,16 @@ public PlayerMoveListener(final Parkour parkour) {
      */
     @EventHandler
     public void onPlayerMove(PlayerMoveEvent event) {
+        Player player = event.getPlayer();
         if (!parkour.getPlayerManager().isPlaying(event.getPlayer())) {
+            if (parkour.getConfig().isVoidDetection()) {
+                if (player.getLocation().getY() < 0) {
```

Comment:
Rather than run this on every move event, I would rather you moved it to the Void damage detection method.